mongoengine/document.py

```diff
@@ -332,6 +332,7 @@ def save(
         _refs=None,
         save_condition=None,
         signal_kwargs=None,
+        upsert=None,
```

Comment:
The default is True so I believe we should make that clear on the api, thus `upsert=True` (similarly to `force_insert=False`)

Reply:
The problem is that is not True.  It only defaults to true IF we do not send a save condition.  I'm not sure how to better fix that because there are 3 states.